src/components/Slider/Rheostat.js

```diff
@@ -0,0 +1,709 @@
+      if (onAfterSet) onAfterSet();
+    });
+  }
+
+  updateNewValues(nextProps) {
+    const { slidingIndex } = this.state;
+
+    // Don't update while the slider is sliding
+    if (slidingIndex !== null) {
+      return;
+    }
+
+    const { max, min, values } = nextProps;
+
+    const nextValues = this.validateValues(values, nextProps);
+
+    this.setState(
+      {
+        handlePos: nextValues.map(value => getPosition(value, min, max)),
+        values: nextValues,
+      },
+      () => this.fireChangeEvent()
+    );
+  }
+
+  render() {
+    const {
+      children,
+      disabled,
+      handle: Handle,
+      max,
+      min,
+      orientation,
+      pitComponent: PitComponent,
+      pitPoints,
+      progressBar: ProgressBar,
+    } = this.props;
+    const { className, handlePos, values } = this.state;
+
+    return (
+      <div
+        className={className}
+        ref={ref => {
+          this.rheostat = ref;
+        }}
+        onClick={!disabled && this.handleClick}
+        style={{ position: 'relative' }}
+      >
+        <div className="rheostat-background" />
+        {handlePos.map((pos, idx) => {
+          const handleStyle =
+            orientation === 'vertical'
+              ? { top: `${pos}%`, position: 'absolute' }
+              : { left: `${pos}%`, position: 'absolute' };
+
+          return (
+            <Handle
+              aria-valuemax={this.getMaxValue(idx)}
+              aria-valuemin={this.getMinValue(idx)}
+              aria-valuenow={values[idx]}
+              aria-disabled={disabled}
+              data-handle-key={idx}
+              className="rheostat-handle"
+              key={`handle-${idx}`}
+              onClick={this.killEvent}
+              onKeyDown={!disabled && this.handleKeydown}
+              onMouseDown={!disabled && this.startMouseSlide}
+              onTouchStart={!disabled && this.startTouchSlide}
+              role="slider"
+              style={handleStyle}
+              tabIndex={0}
+            />
+          );
+        })}
+
+        {handlePos.map((node, idx, arr) => {
+          if (idx === 0 && arr.length > 1) {
+            return null;
+          }
+
+          return (
+            <ProgressBar
+              className="rheostat-progress"
+              key={`progress-bar-${idx}`}
+              style={this.getProgressStyle(idx)}
+            />
+          );
+        })}
+
+        {PitComponent &&
+          pitPoints.map(n => {
+            const pos = getPosition(n, min, max);
+            const pitStyle =
+              orientation === 'vertical'
+                ? { top: `${pos}%`, position: 'absolute' }
+                : { left: `${pos}%`, position: 'absolute' };
+
+            return (
+              <PitComponent key={`pit-${n}`} style={pitStyle}>
+                {n}
+              </PitComponent>
+            );
+          })}
+        {children}
+      </div>
+    );
+  }
+}
+
+Rheostat.propTypes = {
```

Comment:
could this have moved to TS, or do you want to do it in another PR?

Reply:
We can refactor further in other PRs.